### PR TITLE
tests(*) make the `host` and `port` of `grpcbin` configurable

### PR DESCRIPTION
**Pull request by ADD-SP**
### Summary

Now, the `host` and `port` of the `grpcbin` used for testing are fixed in the code and not easy to change. These parameters should be allowed to be changed via environment variables.

This PR introduces three environment variables to improve it:

* `KONG_SPEC_TEST_GRPCBIN_HOST`
* `KONG_SPEC_TEST_GRPCBIN_PORT`
* `KONG_SPEC_TEST_GRPCBIN_SSL_PORT`

### CODE REVIEW

File: spec/helpers.lua [hunk 1]
```diff
@@ -19,6 +19,9 @@ local MOCK_UPSTREAM_PORT = 15555
 local MOCK_UPSTREAM_SSL_PORT = 15556
 local MOCK_UPSTREAM_STREAM_PORT = 15557
 local MOCK_UPSTREAM_STREAM_SSL_PORT = 15558
+local GRPCBIN_HOST = os.getenv("KONG_SPEC_TEST_GRPCBIN_HOST") or "localhost"
+local GRPCBIN_PORT = tonumber(os.getenv("KONG_SPEC_TEST_GRPCBIN_PORT") or "9000")
+local GRPCBIN_SSL_PORT = tonumber(os.getenv("KONG_SPEC_TEST_GRPCBIN_SSL_PORT") or "9001")
```
Comment (bungle) [inline spec/helpers.lua:24]:
```suggestion
local GRPCBIN_PORT = tonumber(os.getenv("KONG_SPEC_TEST_GRPCBIN_PORT")) or 9000
local GRPCBIN_SSL_PORT = tonumber(os.getenv("KONG_SPEC_TEST_GRPCBIN_SSL_PORT")) or 9001
```

**Review by bungle — CHANGES_REQUESTED** [review]
Just a small tweak.

**Review by Tieske — CHANGES_REQUESTED** [review]
the added fields should be added to the generated documentation; https://github.com/Kong/kong/blob/30af56b548bf1bee6a8c493025b5498291de4afc/spec/helpers.lua#L2784

**Review by Tieske — APPROVED** [review]

File: spec/helpers.lua [hunk 2]
```diff
@@ -983,6 +986,7 @@ local function grpc_client(host, port, opts)
       end
 
       local opts = gen_grpcurl_opts(pl_tablex.merge(t.opts, args.opts, true))
+      ngx.log(ngx.ERR, string.format(t.cmd_template, opts, service))
```
Comment (locao) [inline spec/helpers.lua:989]: Is this a forgotten debug message?

Comment (ADD-SP) [reply spec/helpers.lua:989]: Thanks for the reminder, I will remove it.

**Review by gszr — APPROVED** [review]
Nice one! 💯

File: spec/helpers.lua [hunk 3]
```diff
@@ -2797,6 +2800,11 @@ end
 -- @field mock_upstream_stream_port
 -- @field mock_upstream_stream_ssl_port
 -- @field mock_grpc_upstream_proto_path
+-- @field grpcbin_host The host for grpcbin service
+-- @field grpcbin_port The port (SSL disabled) for grpcbin service
+-- @field grpcbin_ssl_port The port (SSL enabled) for grpcbin service
+-- @field grpcbin_url The URL (SSL disabled) for grpcbin service
+-- @field grpcbin_ssl_url The URL (SSL enabled) for grpcbin service
```
Comment (Tieske) [inline spec/helpers.lua:2807]: can we add the name of the env variable to set it to the documentation?

Comment (Tieske) [reply spec/helpers.lua:2807]: probably should be added for the existing entires as well

Comment (ADD-SP) [reply spec/helpers.lua:2807]: Do you mean stating the name of the environment variable in the comment?

Comment (Tieske) [reply spec/helpers.lua:2807]: yes, such that the docs make clear how to set it.

for generating docs do `cd spec && ldoc . && open docs/index.html && cd ..` (requires `ldoc` to be installed)

Comment (ADD-SP) [reply spec/helpers.lua:2807]: done

**Review by Tieske — APPROVED** [review]

File: spec/helpers.lua (hunk 1 above)
Comment (Suika-Kong) [reply spec/helpers.lua]: It has different behavior if the variable has a non-number string value.

**Review by bungle — APPROVED** [review]

**Pull request merged**